Source/WebCore/dom/Observable.cpp

```diff
@@ -25,33 +25,53 @@
 #include "config.h"
 #include "Observable.h"
 
+#include "AbortSignal.h"
 #include "CallbackResult.h"
 #include "Exception.h"
 #include "ExceptionCode.h"
+#include "InternalObserver.h"
+#include "JSSubscriptionObserverCallback.h"
 #include "SubscribeOptions.h"
 #include "Subscriber.h"
 #include "SubscriberCallback.h"
 #include "SubscriptionObserver.h"
-#include "SubscriptionObserverCallback.h"
 
 namespace WebCore {
 
-ExceptionOr<Ref<Observable>> Observable::create(Ref<SubscriberCallback> callback)
+ExceptionOr<Ref<Observable>> Observable::create(RefPtr<SubscriberCallback> callback)
 {
     return adoptRef(*new Observable(callback));
 }
 
 void Observable::subscribe(ScriptExecutionContext& context, std::optional<ObserverUnion> observer, SubscribeOptions options)
+{
+    if (observer.has_value()) {
+        WTF::switchOn(
+            observer.value(),
+            [&](RefPtr<JSSubscriptionObserverCallback>& next) {
+                subscribeInternal(context, adoptRef(*new InternalObserverFromScript(context, next)), options);
```

Comment:
Also, please make a static `InternalObserver::create` function which does the `adoptRef` call (and I would also suggest putting the `suspendIfNeeded` call there instead of in the constructor).

e.g.

```C++
Ref<InternalObserver> InternalObserver::create(<args>)
{
    Ref internalObserver = adoptRef(*new InternalObserver(<args>));
    internalObserver->suspendIfNeeded();
    return internalObserver;
}
```